netsim/modules/vlan.py

```diff
@@ -1303,7 +1324,10 @@ def node_pre_transform(self, node: Box, topology: Box) -> None:
 
         # Set node-level VLAN forwarding mode and merge topology VLAN data into node VLAN data
         set_node_vlan_mode(node.vlans[vname],node,topology.vlans[vname],topology)
+
         node.vlans[vname] = topology.vlans[vname] + node.vlans[vname]
+        if node.vlans[vname].mode == 'route':
```

Comment:
With the updated logic, if the intent is to keep assigning prefixes to node-only vlans then popping the prefix should only be done if this is a global vlan and the mixed_fwd_mode flag is not set

Reply:
Nope, this is OK (although maybe not necessary). We don't need an IRB prefix if the node operates in routed VLAN mode. For global VLANs, the IRB prefix is taken from the global VLAN data.